docs/sources/mimir/send/native-histograms/_index.md

```diff
@@ -0,0 +1,232 @@
+
+  [//]: # "LaTeX equation source: 2^{\\left( 2^{-schema}\right)} <= factor"
+
+  ![Factor equation](factor-equation.svg)
+
+  for example for factor `1.1`:
+
+  [//]: # "Latex equation source: 2^{\\left(2^{-3}\right)}\simeq1.09<=1.1"
+
+  ![Factor 1.1 equation](factor-1.1-equation.svg)
+
+  Table of schema to factor:
+  | _schema_ | _factor_ | | _schema_ | _factor_ |
+  |----------|----------|--|----------|----------|
+  | -4 | 65536 | | 3 | 1.0905 |
+  | -3 | 256 | | 4 | 1.0443 |
+  | -2 | 16 | | 5 | 1.0219 |
+  | -1 | 4 | | 6 | 1.0109 |
+  | 0 | 2 | | 7 | 1.0054 |
+  | 1 | 1.4142 | | 8 | 1.0027 |
+  | 2 | 1.1892 |
+
+- Negative buckets, which contain the coint of observations with a negative value that is smaller or equal to the upper bound and larger than the lower bound of a bucket.
+
+  [//]: # "LaTeX equation source: {-2^{\\left( 2^{-schema}\right)}}^{index+1} < v \leq {-2^{\left( 2^{-schema}\right)}}^{index}"
+
+  ![Negative bucket definition](neg-bucket-def.svg)
+
+  where the `schema` is chosen as above.
+
+## Limiting the number of buckets
+
+The server scraping or receiving native histograms over remote write may limit the number of native histogram buckets it accepts. The server may reject or downscale (reduce resolution and merge adjacent buckets). Even if that wasn't the case, storing and emitting potentially unlimited buckets isn't practical.
+
+The intrumentation libraries of Prometheus have automation to keep the number of buckets down, provided that the maximum bucket number option is used (e.g. `NativeHistogramMaxBucketNumber` in GO).
+
+Once the set maximum is exceeded, the following strategy is enacted:
+
+1. First, if the last reset (or the creation) of the histogram is at least the minimum reset duration ago, then the whole histogram is reset to its initial state (including classic buckets). This only works if the minimum reset duration was set (`NativeHistogramMinResetDuration` in GO).
+
+1. If less time has passed, or if the minimum reset duration is zero, no reset is performed. Instead, the zero threshold is increased sufficiently to reduce the number of buckets to or below the maximum bucket number, but not to more than the maximum zaro threashold (`NativeHistogramMaxZeroThreshold` in GO). Thus, if the threashold is at or above the maximum threashold already nothing happens at this step.
+
+1. After that, if the number of buckets still exceeds maximum bucket number, the resolution of the histogram is reduced by doubling the width of all the sparse buckets (up to a growth factor between one bucket to the next of 2^(2^4) = 65536, see above in [Bucket boundary calculation](#bucket-boundary-calculation)).
+1. Any increased zero threshold or reduced resolution is reset back to their original values once the minimum reset duration has passed (since the last reset or the creation of the histogram).
```

Comment:
Which is still a bug in client_golang, BTW: https://github.com/prometheus/client_golang/issues/1248

(Doesn't have to be included here, just in case somebody asks…)